system/scripts/zone/core/calc_combat.cs

```diff
@@ -73,19 +75,42 @@ public float SCR_CalculateDamage(ICombatEntity attacker, ICombatEntity target, S
 
 		var rnd = RandomProvider.Get();
 
-		var dodgeChance = SCR_GetDodgeChance(attacker, target, skill, skillHitResult);
+		var dodgeChance = SCR_GetDodgeChance(attacker, target, skill, modifier, skillHitResult);
 		if (rnd.Next(100) < dodgeChance)
 		{
 			skillHitResult.Result = HitResultType.Dodge;
 			return 0;
 		}
 
-		var damage = SCR_GetRandomAtk(attacker, target, skill, skillHitResult);
+		var damage = SCR_GetRandomAtk(attacker, target, skill, modifier, skillHitResult);
+
+		// Check for Concentrate
+		if (attacker.Components.Get<BuffComponent>().TryGet(BuffId.Concentrate_Buff, out var concentrateBuff))
+		{
+			var bonusDamage = concentrateBuff.NumArg2;
+			var variableName = "Melia.HitsLeft";
```

Comment:
Would it perhaps make more sense to make this a public const on the buff, so we don't have to copy around a string identifier that is easy to mess up? We could also potentially move the entire logic to a static method on the buff, so all the buff relevant code is in one place, except for where its effects take place. (For now.)

Reply:
You're right, but this is actually a copy-paste error, Concentrate is part of the Swordman PR and shouldn't be here at all (the buffhandler is not present in this PR).  I've removed it.